### ISSUE TITLE
capture offline state and reconnect

### ISSUE THREAD

**Issue by vdelendik** [zebrunner/appium#214]
one of the workaround - reconnect via adb or exit with code 1 to init full device services restart.
modern status-adb with exit code can't be used as state in stf is offline...

reconnecting might be part of the healthcheck for example

```
adb devices
List of devices attached
5200d5e15a71c40b        offline

root@70b3a3c38e53:/home/androidusr# adb reconnect
reconnecting 5200d5e15a71c40b [offline]

root@70b3a3c38e53:/home/androidusr# adb devices
List of devices attached
5200d5e15a71c40b        device
```

**Comment by vdelendik** [comment]
fixed in `appium:2.0.1`. still TBD howto reproduce on demo/stage servers. 
on the production with 100+ devices 2-3 android usually become offline...

**Comment by vdelendik** [comment]
tested successfully on one of the problematic prod device, adjusting healthcheck source based on above commit.
